src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -1839,7 +1839,7 @@ public void visitAssert(JCAssert tree) {
                 JCTree.JCExpression detail = TreeInfo.skipParens(tree.detail);
 
                 pushBody(detail,
-                         FunctionType.functionType(JavaType.J_L_STRING));
+                         FunctionType.functionType(typeToTypeElement(tree.detail.type)));
```

Comment:
Recommend adding another test to the language test for an assert with a second expression evaluating to something other than String.